frame/staking/src/pallet/mod.rs

```diff
@@ -242,7 +243,16 @@ pub mod pallet {
 		/// recorded. This implies that what comes out of iterating this list MIGHT NOT BE AN ACTIVE
 		/// VALIDATOR.
 		type TargetList: SortedListProvider<Self::AccountId, Score = BalanceOf<Self>>;
-
+		/// This is the implementation that's going to maintain approval stake tracking, `VoterList`
+		/// and `TargetList` in near future. Currently running in test mode, which means it's
+		/// maintaining totally detached versions of both lists.
+		/// Once this becomes the source of truth - we should make sure `TargetList` and `VoterList`
+		/// are passed to Staking as `ReadOnlySortedListProvider`.
+		///
+		/// NOTE: Once we have more implementors of EventListener - this could be done similarly to
```

Comment:
Good comment educational-wise, but toward the end it should go away IMO. The fact that `OnStakingUpdate` can be a tuple is abstracted and not something that staking needs to know about. All staking knows is that `OnStakingUpdate` is *something* that will notify anyone who's interested in these updates.

Reply:
I guess this can stay until we switch staking to the new lists.